TrackModel/src/trackmodel/model/Block.java

```diff
@@ -194,32 +168,24 @@ public boolean getBeacon() {
    * This method will return the direction based on an integer.
    * @return An integer value will be returned to indicate the direction
    */
-  public int getDirection() {
-    return direction;
+  public boolean getBiDirection() {
+    return biDirectional;
   }
 
   /**
    * This method will return the next block in the track.
-   * @return A Block will be returned
+   * @return A integer will be returned
    */
-  public Block getNextBlock() {
-    return nextBlock;
+  public int getNextBlock() {
+    return nextBlock1;

```

Comment:
I was not planning on having them stored as a linked list.  I am switching to a hash map.  Also I might not have the information for the next block when I create a block based on what order the data is in in the CSV file, so it will have to be an integer, and then I will get the block.